model_functions.R

```diff
@@ -317,130 +204,62 @@ calc_tilling_impact <- function(tilling_factor = 1,
 }
 
 
-plot_c_stocks <- function(years, 
-                          df,
-                          plot_title = "",
-                          project_name = "test"){
-  
-  df$year <- years
-  df <- df %>% 
-    pivot_longer(cols = c("DPM", "RPM", "BIO", "HUM", "IOM"),
-                 names_to = "soil_comp")
-  
-  
-  plot_title <- paste("Carbon Distribution - ", plot_title)
-  plot <- ggplot(data = df, aes(x = year, y = value, group = soil_comp, colour = soil_comp))+
-    geom_line()+
-    labs(title = plot_title, x = "Time (years)", y = "C Stocks (t/ha)", color = "Soil\nComposition")+
-    theme_classic()+
-    theme(legend.position = "right")
-  
-  
-  if(!dir.exists(file.path("plots", project_name))){dir.create(file.path("plots", project_name))}
-  ggsave(plot, filename = file.path("plots", project_name, paste0(plot_title, ".png")), height = 4.62, width = 5.98)
-  
-}
 
-plot_total_c <- function(years, 
-                         df,
-                         plot_title = "",
-                         project_name = "test"){
-  
-  df <- df %>% rowwise() %>% summarise(c_tot = sum(DPM, RPM, BIO, HUM, IOM), .groups = "drop")
-  df$year <- years
-  
-  plot <- ggplot(data = df, aes(x = year, y = c_tot))+
-    geom_line()+
-    labs(title = plot_title, x = "Time (years)", y = "C Stocks (t/ha)")+
-    theme_classic()
-  
-  plot_title <- paste("Total Carbon - ", plot_title)
-  if(!dir.exists(file.path("plots", project_name))){dir.create(file.path("plots", project_name))}
-  ggsave(plot, filename = file.path("plots", project_name, paste0(plot_title, ".png")), height = 4.62, width = 5.98)
-  
+calc_carbon_over_time <- function(time_horizon = 10, 
+                                  field_carbon_in = c(10,10,10,10,10,10,10,10,10,10), # annual input
+                                  dr_ratios = c(1,1,1,1,1,1,1,1,1), # annual dr ratio
+                                  bare_profile = c(T,T,T,T,T,T,T,T,T,T,T,T), # monthly coverage value
+                                  temp = temp,
+                                  precip = precip,
+                                  evap = evap,
+                                  soil_thick = soil_thick,
+                                  clay = clay,
+                                  pE = pE,
+                                  PS = starting_soil_content,
+                                  tilling_factor = 1){
+  
+  
+  
+  if(length(dr_ratios) != length(field_carbon_in)){stop("Field_carbon_in and dr_ratios should have same length, 1 entry per year")}
+  
+  for (t in 1: time_horizon){
```

Comment:
This for loop might not be necessary since the RothC model can receive arrays / dataframes for the C inputs if they vary over time. Please consider changing.
![image](https://user-images.githubusercontent.com/30375389/128511999-8d2ca9f7-546e-45d2-a782-7fbcfa20766a.png)

Reply:
The challenge with changing this is that the dr value cannot be a data frame and this also theoretically changes over time with differing ratios of manure to crop residue. Otherwise it would make sense.

Reply:
Understood. That's annoying.